src/ComposerEventHandler.php

```diff
@@ -165,12 +164,33 @@ public function onPostAutoloadDump(Event $event): void
 
     private function updateFile(string $source, string $destination): void
     {
-        // TODO: if update happened, do merge here
+        $distDestinationPath = dirname($destination) . '/' . self::DIST_DIRECTORY;
+        $distFilename = $distDestinationPath . '/' . basename($destination);
+
+        $fs = new Filesystem();
+
         if (!file_exists($destination)) {
-            $fs = new Filesystem();
+            // First install config
             $fs->ensureDirectoryExists(dirname($destination));
             $fs->copy($source, $destination);
+        } else {
+            // Update config
+            $sourceContent = file_get_contents($source);
+            $destinationContent = file_get_contents($destination);
+            $distContent = file_exists($distFilename) ? file_get_contents($distFilename) : '';
+
+            if (strcmp($destinationContent, $distContent) === 0) {
+                // Dist file equals with installed config. Installing with overwrite - silent.
+                $fs->copy($source, $destination);
+            } elseif (strcmp($sourceContent, $distContent) !== 0) {
+                // Dist file changed and installed config changed by user.
+                $output = new ConsoleOutput();
+                $output->writeln('<bg=magenta;fg=white>Config file has been changed. Please re-view file: ' . $destination . ' and change with file .dist</>');
+            }
         }
+
+        $fs->ensureDirectoryExists($distDestinationPath);
```

Comment:
We have updated packages in `$this->updates`. Does it worth checking it here to save file operations?